pkg/specgen/generate/container_create.go

```diff
@@ -21,6 +21,27 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
+func platformToCrunVariant(arch string, os string) string {
+	variant := "crun"
+	if arch == "wasm32" || arch == "wasm64" || os == "wasi" {
```

Comment:
Do I need both, IE If a user provides an OS names WASI, I would just expect to use crun-wasm.

Reply:
As of now `OS` and `arch` both are not accepted for `wasi/wasm` but I expect if either one of them is set entrypoint must contain a `wasi` compatible workload.

Reply:
Move this all to containers/common and then make podman simple.